src/reanimated2/js-reanimated/JSReanimated.ts

```diff
@@ -222,6 +227,36 @@ export default class JSReanimated extends NativeReanimated {
       this.platform = Platform.WEB;
     }
   }
+
+  makeSynchronizedDataHolder<T>(
+    _valueRef: ShareableRef<T>
+  ): ShareableSyncDataHolderRef<T> {
+    throw new Error(
+      '[Reanimated] makeSynchronizedDataHolder is not available in JSReanimated.'
+    );
+  }
+
+  getDataSynchronously<T>(_ref: ShareableSyncDataHolderRef<T>): T {
+    throw new Error(
+      '[Reanimated] getDataSynchronously is not available in JSReanimated.'
+    );
+  }
+
+  getViewProp<T>(
+    _viewTag: string,
+    _propName: string,
+    _callback?: (result: T) => void
+  ): Promise<T> {
+    throw new Error(
+      '[Reanimated] getViewProp is not available in JSReanimated.'
+    );
```

Comment:
I'm wondering which approach is better: crashing the user's app or simply calling console.warn. If we call this method on the Web, it means we made a mistake in our logic. However, this method doesn't have any impact on the Web, so it could remain as a no-op. If you think that is better to throw an error I'm ok with that.

Reply:
I think the reason behind it was what you mention - to highlight a possible a mistake in our logic and there's no better way for that than a good ol' error

Reply:
I think at this point we can finally afford to be more strict in terms of error reporting, especially with developer-friendly messages 😄